staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go

```diff
@@ -0,0 +1,462 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package create
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/resource"
+	networkingv1client "k8s.io/client-go/kubernetes/typed/networking/v1"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/scheme"
+	"k8s.io/kubectl/pkg/util"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+var (
+	// Explaining the Regex below:
+	// ^(?P<host>.+) -> Indicates the host - 1-N characters
+	// (?P<path>/.*) -> Indicates the path and MUST start with '/' - / + 0-N characters
+	// Separator from host/path to svcname:svcport -> "="
+	// (?P<svcname>[\w\-]+) -> Service Name (letters, numbers, '-') -> 1-N characters
+	// Separator from svcname to svcport -> ":"
+	// (?P<svcport>[\w\-]+) -> Service Port (letters, numbers, '-') -> 1-N characters
+	regexHostPathSvc = `^(?P<host>.+)(?P<path>/.*)=(?P<svcname>[\w\-]+):(?P<svcport>[\w\-]+)`
+
+	// This Regex is optional -> (....)?
+	// (?P<istls>tls) -> Verify if the argument after "," is 'tls'
+	// Optional Separator from tls to the secret name -> "=?"
+	// (?P<secretname>[\w\-]+)? -> Optional secret name after the separator -> 1-N characters
+	regexTLS = `(,(?P<istls>tls)=?(?P<secretname>[\w\-]+)?)?`
+
+	// The validation Regex is the concatenation of hostPathSvc validation regex
+	// and the TLS validation regex
+	ruleRegex = regexHostPathSvc + regexTLS
+
+	ingressLong = templates.LongDesc(i18n.T(`
+	Create an ingress with the specified name.`))
+
+	ingressExample = templates.Examples(i18n.T(`
+		# Create a catch all ingress pointing to service svc:port and Ingress Class as "otheringress"
+		kubectl create ingress catch-all --class=otheringress --rule="_/=svc:port"
```

Comment:
Is the _ really needed?  `host.com/foo` and `/foo` are not ambiguous...

Reply:
The idea of _ was exactly to remove the ambiguity before I started using the regex. But yes, this might be an improvement. Putting in my TODO for tomorrow, to check the impact on simply using "/foo" directly and verifying in the code if the len of the hostpath array is bigger than 1 then consider the first field as host and the second as path, otherwise just path :)

Reply:
@thockin created a new PR to solve this (https://github.com/kubernetes/kubernetes/pull/95660) and also another...well... bug because in the final past I've removed the generators flag but forgot to add the dryrun flag and now the command simply panics